pkg/config/config.go

```diff
@@ -654,7 +654,13 @@ func (c *baseConfig) resolve(builder pathBuilder, fsys fs.FS) error {
 	// Resolve functions config
 	for slug, function := range c.Functions {
 		if len(function.Entrypoint) == 0 {
-			function.Entrypoint = filepath.Join(builder.FunctionsDir, slug, "index.ts")
+			indexEntrypoint := filepath.Join(builder.FunctionsDir, slug, "index.ts")
+			mainEntrypoint := filepath.Join(builder.FunctionsDir, slug, "main.ts")
+			if _, err := fs.Stat(fsys, indexEntrypoint); err == nil {
+				function.Entrypoint = indexEntrypoint
+			} else if _, err := fs.Stat(fsys, mainEntrypoint); err == nil {
+				function.Entrypoint = mainEntrypoint
+			}
```

Comment:
```suggestion
			function.Entrypoint = filepath.Join(builder.FunctionsDir, slug, "index.ts")
			if _, err := fs.Stat(fsys, function.Entrypoint); errors.Is(err, os.ErrNotExist) {
				mainEntrypoint := filepath.Join(builder.FunctionsDir, slug, "main.ts")
				if _, err := fs.Stat(fsys, mainEntrypoint); err == nil {
					function.Entrypoint = mainEntrypoint
				}
			}
```

We should still pass down default entrypoint path.